File: src/main/java/utilities/Timer.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package utilities;

import interfaces.Updateable;

/**
 *
 * @author brooks42
 */
public class Timer implements Updateable {

    // the time to wait before this Timer's tick() method is called
    private long waitTime = 0l;
    // the amount of time that has passed so far
    private long passedTime = 0l;

    /**
     * Creates a new Timer with the passed waitTime.
     *
     * @param waitTime the time (in milliseconds) before this time will call its
     * tick() method.
     */
    public Timer(long waitTime) {
        this.waitTime = waitTime;
    }

    @Override
    public void update(float tpf) {
        passedTime += tpf;
        if (passedTime >= waitTime) {
            tick();
        }
    }
    
    /**
     * Resets this Timer. This is useful for calling during the tick() method, 
     * if you'd like the timer to repeat.
     */
    public void reset(){
        this.passedTime = 0l;
    }
    
    /**
     * This Timer object's tick() method. 
     * 
     * You should create anonymous Timer classes that override this click, much
     * in the same way you create separate threads (although this will be in the
     * main thread)
     */
    public void tick(){
        
    }
}

File: src/main/java/utilities/Timer.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package utilities

import interfaces.Updateable

/**
 * Discussion: at the moment this is completely unused but as the codebase matures I think it'll be a super useful class,
 * I like the idea of using it to build out animations or purely just events, stuff like damage over time etc
 * Need to rethink its usefulness and architecture
 *
 * @author brooks42
 */
class Timer(waitTime: Long) : Updateable {

    // the time to wait before this Timer's tick() method is called
    private val waitTime = 0L

    // the amount of time that has passed so far
    private var passedTime = 0L
    override fun update(tpf: Float) {
        passedTime += tpf.toLong()
        if (passedTime >= waitTime) {
            tick()
        }
    }

    /**
     * Resets this Timer. This is useful for calling during the tick() method,
     * if you'd like the timer to repeat.
     */
    fun reset() {
        passedTime = 0L
    }

    /**
     * This Timer object's tick() method.
     *
     * You should create anonymous Timer classes that override this click, much
     * in the same way you create separate threads (although this will be in the
     * main thread)
     */
    fun tick() { }
}